rotkehlchen/chain/arbitrum_one/modules/umami/utils.py

```diff
@@ -0,0 +1,54 @@
+
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.arbitrum_one.modules.umami.constants import UMAMI_ASSET_VAULT_ABI
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import EvmToken
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.inquirer import Inquirer
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def get_umami_vault_token_price(
+        inquirer: 'Inquirer',
+        vault_token: 'EvmToken',
+        evm_inquirer: 'EvmNodeInquirer',
+) -> Price | None:
+    """Gets token price for Umami vault.
+
+    Multiplies vault's price per share by the underlying token's USD price.
+
+    Args:
+        inquirer: For underlying token price lookup.
```

Comment:
again same thing here. Really no need for such huge docstrings with the arguments of the function